sdk/storage/Azure.Storage.Common/tests/Shared/TestExtensions.cs

```diff
@@ -61,5 +64,125 @@ public static Uri ToHttp(this Uri uri)
+                            conn.QueueStorageUri.SecondaryUri.ToString());
+                    }
+
+                    if (conn.TableStorageUri.SecondaryUri != null)
+                    {
+                        conn.Settings.Add(Constants.ConnectionStrings.TableSecondaryEndpointSetting,
+                            conn.TableStorageUri.SecondaryUri.ToString());
+                    }
+
+                    if (conn.FileStorageUri.SecondaryUri != null)
+                    {
+                        conn.Settings.Add(Constants.ConnectionStrings.FileSecondaryEndpointSetting,
+                            conn.FileStorageUri.SecondaryUri.ToString());
+                    }
+                }
+            }
+
+            var listOfSettings = conn.Settings.Select(pair => string.Format(CultureInfo.InvariantCulture, "{0}={1}", pair.Key, pair.Value)).ToList();
+
+            if (conn.Credentials != null && !conn.IsDevStoreAccount)
+            {
+                listOfSettings.Add(ToString(conn.Credentials, exportSecrets));
+            }
+
+            if (!string.IsNullOrWhiteSpace(conn._accountName) && (conn.Credentials is StorageSharedKeyCredential sharedKeyCredentials ? string.IsNullOrWhiteSpace(sharedKeyCredentials.AccountName) : true))
+            {
+                listOfSettings.Add(string.Format(CultureInfo.InvariantCulture, "{0}={1}", Constants.ConnectionStrings.AccountNameSetting, conn._accountName));
+            }
+
+            return string.Join(";", listOfSettings);
+        }
+
+        private static string ToString(object credentials, bool exportSecrets)
+        {
+            if (credentials is StorageSharedKeyCredential sharedKeyCredentials)
+            {
+                return string.Format(
+                    CultureInfo.InvariantCulture,
+                    "{0}={1};{2}={3}",
+                    Constants.ConnectionStrings.AccountNameSetting,
+                    sharedKeyCredentials.AccountName,
+                    Constants.ConnectionStrings.AccountKeySetting,
+                    exportSecrets ? ((StorageSharedKeyCredential)credentials).ExportBase64EncodedKey() : "Sanitized");
+            }
+            else if (credentials is SharedAccessSignatureCredentials sasCredentials)
+            {
+                return string.Format(CultureInfo.InvariantCulture, "{0}={1}", Constants.ConnectionStrings.SharedAccessSignatureSetting, exportSecrets ? sasCredentials.SasToken : "[signature hidden]");
+            }
+
+            return string.Empty;
+        }
+
+        internal static string ExportBase64EncodedKey(this StorageSharedKeyCredential credential)
+        {
+            byte[] key = credential.GetAccountKey();
+            return key == null ?
+                null :
+                Convert.ToBase64String(key);
+        }
+
+        internal static byte[] GetAccountKey(this StorageSharedKeyCredential credential)
+        {
+            Type type = credential.GetType();
```

Comment:
Add a quick comment about why we're using private reflection along the lines of `// We don't want to expose the AccountKey to users to encourage them to properly manage their secrets and we don't want to expose all of Azure.Storage.Common's internals to all our tests, so we're making a strategic choice to use private reflection for things field` or similar